File: backend/src/main/java/by/bsu/cardirectory/entity/Car.java
package by.bsu.cardirectory.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.ToString;
import org.hibernate.annotations.OnDelete;

import java.math.BigDecimal;
import java.util.Date;

@Entity
@Data
public class Car {

    @Id
    @SequenceGenerator(
            allocationSize = 1,
            name = "car_id_seq",
            sequenceName = "car_id_seq"
    )
    @GeneratedValue(
            strategy = GenerationType.SEQUENCE,
            generator = "car_id_seq"
    )
    private Long id;

    private String model;

    private String description;

    private Date releaseDate;

    private Double fuelEfficiency;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "manufacturer_id")
    @ToString.Exclude
    private Manufacturer manufacturer;
}

File: frontend/src/interfaces/car.ts
import { IEntity } from './entity';
import { IManufacturer } from './manufacturer';

export interface ICar extends IEntity {
  model: string;
  description: string | null;
  releaseDate: string;
  fuelEfficiency: number;
  manufacturer?: IManufacturer | null;
}

export type ICarValues = ICar[keyof ICar];

export interface ICarPayload {
  model: string;
  description: string | null;
  releaseDate: string;
  fuelEfficiency: number;
  manufacturer?: string | null;
}
